src/main.rs

```diff
@@ -292,17 +292,19 @@ async fn main() -> Result<(), String> {
                     data.cmd.handle(s, command);
                 }
             } else {
-                let parsed = command::parse(cmd_without_prefix);
-                if let Some(commands) = parsed {
-                    if let Some(data) = s.user_data::<UserData>().cloned() {
-                        for cmd in commands {
-                            data.cmd.handle(s, cmd);
+                match command::parse(cmd_without_prefix) {
+                    Ok(commands) => {
+                        if let Some(data) = s.user_data::<UserData>().cloned() {
+                            for cmd in commands {
+                                data.cmd.handle(s, cmd);
+                            }
                         }
                     }
-                } else {
-                    let mut main = s.find_name::<ui::layout::Layout>("main").unwrap();
-                    let err_msg = format!("Failed to parse command(s): \"{}\"", cmd_without_prefix);
-                    main.set_result(Err(err_msg));
+                    Err(err) => {
+                        let mut main = s.find_name::<ui::layout::Layout>("main").unwrap();
+                        let err_msg = format!("Failed to parse command(s): {}", err);
+                        main.set_result(Err(err_msg));
+                    }
```

Comment:
This has the occasional problem where the error message's length exceeds the width of the terminal window, causing part of the message to be missing. Is there anything we can do about this?

Reply:
I got rid of some wordiness, but the problem is not solved at the root. I'm thinking of some kind of auto-scroll or multi-line message.

Reply:
I took another look today after simply shortening the message and, eh, it's good enough. Too much work than it's worth for now.